avocado/core/nrunner.py

```diff
@@ -27,6 +27,12 @@
 #: SpawnMethod.STANDALONE_EXECUTABLE compatible spawners
 RUNNERS_REGISTRY_STANDALONE_EXECUTABLE = {}
 
+#: All known runner Python classes.  This is a dictionary keyed by a
+#: runnable kind, and value is a class that inherits from
+#: :class:`BaseRunner`.  Suitable for spawners compatible with
+#: SpawnMethod.PYTHON_CLASS
+RUNNERS_REGISTRY_PYTHON_CLASS = {}
```

Comment:
For sure this is a big improvement for the current implementation. Thanks for that.

One disadvantage that I can see here, is that this "registration" is a little "loose": Thinking about the future where runners can be developed by a third-party, they could change other runners registrations by messing up with this dict.

But I understand that this could be improved later and/or is not a problem for now. So I'm cool with that.